clients/python/README.md

```diff
@@ -178,6 +178,92 @@ By default, all queries will be `ascending`, but this method is also available f
 > Advanced usage note: You can also set the `page_size()` that you want the Pager to use when invoking the Model Registry backend.
 > When using it as an iterator, it will automatically manage pages for you.
 
+### Uploading local models to external storage and registering them
+
+To both upload and register a model, use the convenience method `upload_artifact_and_register_model`.
+
+This method supports both s3-based storage (via [boto3](https://github.com/boto/boto3)) as well as OCI-based image registries (via [olot](https://github.com/containers/olot), using either of the CLI tools [skopeo](https://github.com/containers/skopeo) or [oras](https://github.com/oras-project/oras))
+
+In order to utilize this method you must instantiate an `upload_params` object which contains the necessary locations and credentials needed to perform the upload to that storage provider.
+
+#### S3 based external storage
+
+```python
+s3_upload_params = S3Params(
+    bucket="my-bucket",
+    s3_prefix="hello_world_model",
+    access_key_id="my-key-id",
+    access_key_secret="super-secret",
+    region="us-east-1"
+)
+
+registered_model = client.upload_artifact_and_register_model(
+    name="hello_world_model",
+    model_fiels_path="/home/user-01/models/model_training_01"
+    # If the model consists of a single file, such as a .onnx file, you can specify that as well
+    # model_fiels_path="/home/user-01/models/model_training_01.onnx"
+    author="Mr. Trainer",
+    version="0.0.1",
+    upload_params=s3_upload_params
+)
+```
+
+#### OCI-registry based storage
+
+```python
+oci_upload_params = OCIParams(
+    base_image="busybox",
+    oci_ref="registry.example.com/acme_org/hello_world_model:0.0.1"
```

Comment:
very fair use of Tags, when we will have permanent Tags in some OCI registry we will update here as well 🚀